### PR TITLE
chore: add Page.reload for BiDi

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: test/TestExpectations.json [hunk 1]
```diff
@@ -1367,6 +1361,12 @@
     "parameters": ["firefox", "webDriverBiDi"],
     "expectations": ["FAIL"]
   },
+  {
+    "testIdPattern": "[navigation.spec] navigation Page.goto should return last response in redirect chain",
```
Comment (Lightning00Blade) [inline test/TestExpectations.json:1377]: That is due to a fix in Mapper that now handles correctly `ExtraInfo` events for Redirects, but we still are missing the full support there so only the first request is reported, not the last one as expected in test.

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "test:chrome:new-headless": "wireit",
     "test:chrome:headless": "wireit",
     "test:chrome:bidi": "wireit",
+    "test:chrome:bidi-local": "wireit",
```
Comment (Lightning00Blade) [inline package.json:33]: Added for ease of use as BiDi requires different browser version. Will look at a way to unify both command and CI.

File: packages/puppeteer-core/src/common/bidi/Page.ts [hunk 3]
```diff
@@ -313,6 +313,22 @@ export class Page extends PageBase {
...
+    options?: WaitForOptions
+  ): Promise<HTTPResponse | null> {
+    const [response] = await Promise.all([
+      this.waitForResponse(response => {
+        return (
+          response.request().isNavigationRequest() &&
+          response.url() === this.url()
+        );
+      }),
+      this.mainFrame().context().reload(options),
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/bidi/Page.ts:326]: let's use `this.mainFrame().reload()` ?

File: packages/puppeteer-core/src/common/bidi/BrowsingContext.ts [hunk 4]
```diff
@@ -116,6 +117,26 @@ export class BrowsingContext extends EventEmitter {
     }
   }
 
+  async reload(options: WaitForOptions = {}): Promise<void> {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/bidi/BrowsingContext.ts:120]: does it only work on frames?

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/bidi/BrowsingContext.ts:120]: For what else are we expecting it to work. In BiDi spec it's any browsing context.

File: packages/puppeteer-core/src/common/bidi/Page.ts (hunk 3 above)
Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/bidi/Page.ts:326]: This mistake on my part. The spec clearly defines any BrowsingContext not only the `TopLevel one` aka `Page` as WebDriver.

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/bidi/Page.ts:326]: What I said is true, but I think we want to keep the Puppeteer implementation the same for as per the Docs.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**